ONSdigital/dp-compose
# Feature/add frontend services to cantabular journey

# What
Added the front-end services required to trigger the Cantabular dataset import to the compose journey.

# How to review

Pull the branch `feature/add-local-dockerfile` for the following services:
`dp-api-router`
`dp-frontend-dataset-controller`
`dp-frontend-router`
`dp-publishing-dataset-controller`
`florence`

These all have active PRs as well and it would be easiest to review all at the same time as they are all interdependent on each-other.

Make sure to follow instructions in the `# Requirements` section of the README.md for dp-compose/cantabular-import closely and run `make start` to bring up the services.

To test an import journey take the following steps:

- Visit Florence in your browser on http://localhost:8081/florence
- Navigate to the Datasets tab and find the Cantabular Example 1 recipe.
- Proceed with triggering an import by hitting save and continue -> 2021 -> Submit To Publishing
- Once the files have successfully imported you will receive a confirmation message on the following screen.
-  Navigate to Colletions and create a new collection.
-  Once created, click on your new collection and choose create/edit content on the right. Choose filterable dataset
- On the next page you need to choose Create New Dataset followed by the Cantabular Dataset you just uploaded
- You then be able to create new edition and will be bought to the dataset metadata page. Choose a publish date and the rest is optional.
- Hit save and submit for review.
- At this point you will need to log in as a different user, navigate to your collections and review, submit for approval and approve.
- Now your dataset will be ready to publish from the publishing queue tab along the top.
- Attempting to publish the dataset will appear to fail, but you will be able to confirm it has been created by visiting http://localhost:8081/datasets/cantabular-example-1.

This is expected behaviour and confirms the docker-compose journey is working as expected!
